ui/widgets/tabs.js

```diff
@@ -270,7 +271,19 @@ $.widget( "ui.tabs", {
 			return index;
 		}
 
-		while ( $.inArray( constrain(), this.options.disabled ) !== -1 ) {
+		function canActivate( index ) {
+			if ( $.inArray( constrain(), that.options.disabled ) !== -1 ) {
+				return false;
+			}
+
+			if ( that.tabs.eq( index ).is( ":hidden" ) ) {
```

Comment:
Just a thought but the `:visible` and `:hidden` selectors are pretty slow and not 100% reliable. We don't document that hidden tabs are automatically not navigable. Would it perhaps be better to leave this change out and document that you should mark hidden tabs as disabled?